### PR TITLE
Add jobs to run tests with pg 12

### LINKED ISSUE #3034: Add Postgres 12 jobs
Currently we are testing master with postgres 10 and 11. We should add postgres 12 to our tests.

### LINKED ISSUE #3061: Make placement order consistent during writes
While making citus work with pg12 we ran into an issue that is suspected to be caused by an inconsistent placement order for tables with `shard_replication_factor` > 1.

Failing test suite `with_transaction` in non-adaptive mode.
Query:
```sql
-- multi-statement multi shard modifying statements should work
BEGIN;
	SELECT count (*) FROM second_raw_table;
 count 
-------
   101
(1 row)

	WITH distinct_count AS (
		SELECT count(DISTINCT created_at) FROM raw_table
	),
	ids_inserted AS
	(
		INSERT INTO raw_table VALUES (11, 1000, now()) RETURNING tenant_id
	)
	UPDATE raw_table SET created_at = '2001-02-10 20:00:00'
	WHERE tenant_id IN (SELECT tenant_id FROM ids_inserted) AND tenant_id < (SELECT count FROM distinct_count);
DEBUG:  generating subplan 9_1 for CTE distinct_count: SELECT count(DISTINCT created_at) AS count FROM with_transactions.raw_table
DEBUG:  generating subplan 9_2 for CTE ids_inserted: INSERT INTO with_transactions.raw_table (tenant_id, income, created_at) VALUES (11, 1000, now()) RETURNING tenant_id
DEBUG:  Plan 9 query after replacing subqueries and CTEs: UPDATE with_transactions.raw_table SET created_at = 'Sat Feb 10 20:00:00 2001 PST'::timestamp with time zone WHERE ((tenant_id OPERATOR(pg_catalog.=) ANY (SELECT ids_inserted.tenant_id FROM (SELECT intermediate_result.tenant_id FROM read_intermediate_result('9_2'::text, 'binary'::citus_copy_format) intermediate_result(tenant_id integer)) ids_inserted)) AND (tenant_id OPERATOR(pg_catalog.<) (SELECT distinct_count.count FROM (SELECT intermediate_result.count FROM read_intermediate_result('9_1'::text, 'binary'::citus_copy_format) intermediate_result(count bigint)) distinct_count)))
+ERROR:  cannot establish a new connection for placement 800007, since DML has been executed on a connection that is in use
	TRUNCATE second_raw_table;
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
COMMIT;
```

For now we have punted on this issue by going back to `replication_factor = 1` during this test.

### PR DESCRIPTION
**Pull request by SaitTalhaNisanci**
DESCRIPTION: Add postgres 12 jobs

This PR adds postgres12 tests. I have created another image with postgres 12 installed. Currently some of the tests fail in my local too with postgres 12.

Fixes #3034.
Fixes #3061.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -207,39 +207,168 @@ jobs:
...
+          flags: 'test_12,isolation'        
+
 workflows:
   version: 2
   build_and_test:
     jobs:
       - build
-      - check-style
-      - check-sql-snapshots
+      # - check-style
```
[the remark this answers is not in the document]
Comment (SaitTalhaNisanci) [reply .circleci/config.yml]: That was just to run less jobs to not block other people's tests. I will enable them again.

[the remark this answers is not in the document]
Comment (SaitTalhaNisanci) [reply .circleci/config.yml]: @thanodnl I enabled all the jobs.

**Comment by serprex** [comment]
postgres should be built with `--with-icu` for better collation support

I've also gotten the floating errors. It's necessary that tests be run with `extra_float_digits=0`, in pg_regress_multi.pl there's a line that tries to do that, but might not be applying to check-vanilla

File: .circleci/config.yml [hunk 2]
```diff
@@ -228,6 +162,115 @@ jobs:
...
+      - run: 
+          name: 'Install and Test (fol)'
+          command: 'chown -R circleci:circleci /home/circleci && install-and-test-ext check-follower-cluster'
+      - run:
+          command: |
+            mkdir -p /tmp/core_dumps
+            cp core.* /tmp/core_dumps
+          when: on_fail
+      - codecov/upload: 
+          flags: 'follower'
```
Comment (thanodnl) [inline .circleci/config.yml]: missing version 12 flag

**Pull request merged**